src/silo/preprocessing/metadata_info.cpp

```diff
@@ -9,137 +9,173 @@
-   const silo::config::DatabaseConfig& database_config
-) {
-   std::vector<std::string> config_metadata_fields;
-   std::transform(
-      database_config.schema.metadata.begin(),
-      database_config.schema.metadata.end(),
-      std::back_inserter(config_metadata_fields),
-      [](auto metadata) { return metadata.name; }
-   );
-
-   std::unordered_map<std::string, std::string> validated_metadata_fields;
-   for (const auto& [field_name, access_path] : found_metadata_fields) {
-      if (std::find(config_metadata_fields.begin(), config_metadata_fields.end(), field_name) !=
-          config_metadata_fields.end()) {
-         validated_metadata_fields.emplace(field_name, access_path);
-      } else {
-         SPDLOG_WARN(
-            "Metadata field {} ({}), which is contained in the file is not contained in the "
-            "config.",
-            field_name,
-            access_path
-         );
-      }
+using silo::config::ValueType;
+
+std::string toSQLType(ValueType value_type) {
+   switch (value_type) {
+      case ValueType::INT:
+         return "INT4";
+      case ValueType::STRING:
+      case ValueType::PANGOLINEAGE:
+         return "VARCHAR";
+      case ValueType::FLOAT:
+         return "FLOAT4";
+      case ValueType::BOOL:
+         return "BOOL";
+      case ValueType::DATE:
+         return "DATE";
    }
-   for (const std::string& name : config_metadata_fields) {
-      if (!validated_metadata_fields.contains(name)) {
-         throw silo::preprocessing::PreprocessingException(fmt::format(
-            "The metadata field '{}' which is contained in the database config is "
-            "not contained in the input.",
-            name
-         ));
-      }
-   }
-
-   std::string metadata_field_string;
-   for (const auto& [field_name, select] : validated_metadata_fields) {
-      metadata_field_string += "'";
-      metadata_field_string += field_name;
-      metadata_field_string += "' with selection '";
-      metadata_field_string += select;
-      metadata_field_string += "',";
-   }
-   SPDLOG_TRACE("Found metadata fields: " + metadata_field_string);
-   return validated_metadata_fields;
+   abort();
 }
 
 }  // namespace
 
 namespace silo::preprocessing {
 
-MetadataInfo::MetadataInfo(std::unordered_map<std::string, std::string> metadata_selects)
-    : metadata_selects(std::move(metadata_selects)) {}
-
-MetadataInfo MetadataInfo::validateFromMetadataFile(
+void MetadataInfo::validateMetadataFile(
    const std::filesystem::path& metadata_file,
    const silo::config::DatabaseConfig& database_config
 ) {
    duckdb::DuckDB duck_db(nullptr);
    duckdb::Connection connection(duck_db);
    // Get the column names (headers) of the table
-   auto result =
-      connection.Query(fmt::format("SELECT * FROM '{}' LIMIT 0", metadata_file.string()));
+   auto result = connection.Query(fmt::format(
+      "SELECT * FROM read_csv_auto('{}', delim = '\t', header = true) LIMIT 0",
```

Comment:
What's actually the reason for using `read_csv_auto` instead of putting the plain filename there?

Reply:
It is more explicit. Also it prevents some errors where the auto-detection cannot infer that there is a header file, or erroneously assumes ',' as delimiter and have all (tab-separated) columns in a single column instead